modules/tinymce/src/core/main/ts/html/InvalidNodes.ts

```diff
@@ -178,7 +178,7 @@ const isInvalid = (schema: Schema, node: AstNode, parent: AstNode | null | undef
 
   // heading element is valid if it is the only one child of summary
   if (AstNodeType.isSummary(parent) && AstNodeType.isHeading(node)) {
-    return !hasOnlyChild(parent, node.name);
+    return !(parent?.firstChild === node && parent?.lastChild === node);
```

Comment:
Right so the parent here is not a direct parent but the closest valid parent element and the node is the invalid node within that context. Maybe we should add some form of comment that explains this better. I think I was assuming that the parent here would be the parent of the node in this case and that is why I wrongly suggested that we could use the `hasOnlyChild` function.